The AWS libcrypto Rust crate is being packaged in Fedora but stalled due to the
additional cryptographic requirements.  Firecracker only uses it to get some
random bytes, so this reverts 246a308ef6d1054f7551ebeee7f0191b8518d2ad to go
back to the original implementation with the rand crate instead.

--- a/resources/seccomp/aarch64-unknown-linux-musl.json
+++ b/resources/seccomp/aarch64-unknown-linux-musl.json
@@ -111,7 +111,7 @@
             },
             {
               "syscall": "getrandom",
-              "comment": "getrandom is used by aws-lc library which we consume in virtio-rng"
+              "comment": "getrandom is used by virtio-rng to initialize the rand crate"
             },
             {
                 "syscall": "accept4",
@@ -213,7 +213,16 @@
             },
             {
                 "syscall": "madvise",
-                "comment": "Used by the VirtIO balloon device and by musl for some customer workloads. It is also used by aws-lc during random number generation. They setup a memory page that mark with MADV_WIPEONFORK to be able to detect forks. They also call it with -1 to see if madvise is supported in certain platforms." 
+                "comment": "Used by the VirtIO balloon device and by musl for some customer workloads",
+                "args": [
+                    {
+                        "index": 2,
+                        "type": "dword",
+                        "op": "eq",
+                        "val": 4,
+                        "comment": "libc::MADV_DONTNEED"
+                    }
+                ]
             },
             {
                 "syscall": "mmap",
--- a/resources/seccomp/x86_64-unknown-linux-musl.json
+++ b/resources/seccomp/x86_64-unknown-linux-musl.json
@@ -111,7 +111,7 @@
             },
             {
                 "syscall": "getrandom",
-                "comment": "getrandom is used by aws-lc library which we consume in virtio-rng"
+                "comment": "getrandom is used by virtio-rng to initialize the rand crate"
             },
             {
                 "syscall": "accept4",
@@ -213,7 +213,16 @@
             },
             {
                 "syscall": "madvise",
-                "comment": "Used by the VirtIO balloon device and by musl for some customer workloads. It is also used by aws-lc during random number generation. They setup a memory page that mark with MADV_WIPEONFORK to be able to detect forks. They also call it with -1 to see if madvise is supported in certain platforms."
+                "comment": "Used by the VirtIO balloon device and by musl for some customer workloads",
+                "args": [
+                    {
+                        "index": 2,
+                        "type": "dword",
+                        "op": "eq",
+                        "val": 4,
+                        "comment": "libc::MADV_DONTNEED"
+                    }
+                ]
             },
             {
                 "syscall": "mmap",
--- a/src/vmm/Cargo.toml
+++ b/src/vmm/Cargo.toml
@@ -19,7 +19,7 @@
 aes-gcm = { version = "0.10.1", default-features = false, features = ["aes"] }
 anyhow = "1.0.98"
 arrayvec = { version = "0.7.6", optional = true }
-aws-lc-rs = { version = "1.13.3", features = ["bindgen"] }
+rand_core = { version = "0.9.3", features = ["os_rng"] }
 base64 = "0.22.1"
 bincode = { version = "2.0.1", features = ["serde"] }
 bitflags = "2.9.1"
--- a/src/vmm/src/devices/acpi/vmgenid.rs
+++ b/src/vmm/src/devices/acpi/vmgenid.rs
@@ -2,9 +2,8 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use acpi_tables::{Aml, aml};
-use aws_lc_rs::error::Unspecified as RandError;
-use aws_lc_rs::rand;
 use log::{debug, error};
+use rand_core::{OsError, OsRng, TryRngCore};
 use serde::{Deserialize, Serialize};
 use vm_memory::{GuestAddress, GuestMemoryError};
 use vm_superio::Trigger;
@@ -45,7 +44,7 @@
     /// Error accessing VMGenID memory: {0}
     GuestMemory(#[from] GuestMemoryError),
     /// Create generation ID error: {0}
-    GenerationId(#[from] RandError),
+    GenerationId(#[from] OsError),
     /// Failed to allocate requested resource: {0}
     Allocator(#[from] vm_allocator::Error),
 }
@@ -100,9 +99,9 @@
     }
 
     // Create a 16-bytes random number
-    fn make_genid() -> Result<u128, RandError> {
+    fn make_genid() -> Result<u128, OsError> {
         let mut gen_id_bytes = [0u8; 16];
-        rand::fill(&mut gen_id_bytes)
+        OsRng.try_fill_bytes(&mut gen_id_bytes)
             .inspect_err(|err| error!("vmgenid: could not create new generation ID: {err}"))?;
         Ok(u128::from_le_bytes(gen_id_bytes))
     }
--- a/src/vmm/src/devices/virtio/rng/device.rs
+++ b/src/vmm/src/devices/virtio/rng/device.rs
@@ -5,7 +5,7 @@
 use std::ops::Deref;
 use std::sync::Arc;
 
-use aws_lc_rs::rand;
+use rand_core::{OsError, OsRng, TryRngCore};
 use log::info;
 use vm_memory::GuestMemoryError;
 use vmm_sys_util::eventfd::EventFd;
@@ -35,7 +35,7 @@
     /// Bad guest memory buffer: {0}
     GuestMemory(#[from] GuestMemoryError),
     /// Could not get random bytes: {0}
-    Random(#[from] aws_lc_rs::error::Unspecified),
+    Random(#[from] OsError),
     /// Underlying IovDeque error: {0}
     IovDeque(#[from] IovDequeError),
 }
@@ -120,7 +120,7 @@
         }
 
         let mut rand_bytes = vec![0; self.buffer.len() as usize];
-        rand::fill(&mut rand_bytes).inspect_err(|_| {
+        OsRng.try_fill_bytes(&mut rand_bytes).inspect_err(|_| {
             METRICS.host_rng_fails.inc();
         })?;
 
